### PR TITLE
Our team cubit

### LINKED ISSUE #53: Our Team Screen -> Cubit
### Is your feature request related to a problem? Please describe.

We need to design the cubit for Our Team Screen.

### Describe the solution you'd like

1. should have states: loading, success, error.
2. fake api repository.
3. tests.

### PR DESCRIPTION
**Pull request by divy-arpit**
### Description

This PR adds cubit for our team screen

**Type of Change:** (Delete irrelevant options)
- Code

**Code/Quality Assurance Only** (Delete irrelevant options)

- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

Tested on local machine.

### Checklist:

- [X] My PR follows the style guidelines of this project
- [X] I have performed a self-review of my own code or materials
- [X] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] Any dependent changes have been merged
- [X] I have added one/more required dependencies
- [X] I have formatted the code using dart format (dartfmt)

**Code/Quality Assurance**
- [X] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

**Mention the dependencies/packages/plugins that have been added (if any)**

### CODE REVIEW

File: lib/screens/our_team/out_team_repository.dart [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            "modified_at": "2021-01-13T11:43:16.131751Z",
+            "name": "absan",
+            "profile_pic": "/media/static/uploads/speakers/Ansh.jpg",
+          }
+        ],
+        "message": "Member Fetched Successfully"
+      };
+      List<Member> memberList = List();
+      (response["data"] as List).map((e) => memberList.add(Member.fromJson(e))).toList();
+      return memberList;
```
Comment (TreavVasu) [inline lib/screens/our_team/out_team_repository.dart]: Instead  of this from Line 33 to 35 do this 

      return (response["data"] as List).map((e) => Member.fromJson(e)).toList();

File: lib/screens/our_team/our_team.dart [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    return Container();
+  }
+
+  Widget _buildSuccess(BuildContext context, List<Member> speakerList) {
+    //TODO: UI
+    return Center(
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: <Widget>[
+          Text(speakerList[0].name, textAlign: TextAlign.center),
```
Comment (TreavVasu) [inline lib/screens/our_team/our_team.dart]: You probably wanna change that with TeamList or so

**Review by TreavVasu — CHANGES_REQUESTED** [review]
LGTM other than these mentioned changes.

File: lib/screens/our_team/cubit/our_team_state.dart [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+class OurTeamInitial extends OurTeamState {
+  const OurTeamInitial();
+}
+
+class OurTeamLoading extends OurTeamState {
+  const OurTeamLoading();
+}
+
+class OurTeamSuccess extends OurTeamState {
+  final List<Member> memberList;
```
Comment (werainkhatri) [inline lib/screens/our_team/cubit/our_team_state.dart:20]: There should be List of MemberCategory here.

**Comment by Hard-Coder05** [comment]
@divy-arpit @TreavVasu Is this completed? @werainkhatri please review this and close this one

File: lib/screens/our_team/out_team_repository.dart [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            "flag": true,
+            "id": 1,
+            "modified_at": "2021-01-13T11:43:16.131751Z",
+            "name": "absan",
+            "profile_pic": "/media/static/uploads/speakers/Ansh.jpg",
+          }
+        ],
+        "message": "Member Fetched Successfully"
+      };
+      return (response["data"] as List).map((e) => Member.fromJson(e)).toList();
```
Comment (werainkhatri) [inline lib/screens/our_team/out_team_repository.dart:33]: here the members should be segregated into categories and then return the list of category.

**Review by werainkhatri — CHANGES_REQUESTED** [review]
Coordinate with the backend team on structure of the api.

File: lib/screens/our_team/out_team_repository.dart (hunk 1 above)
Comment (werainkhatri) [reply lib/screens/our_team/out_team_repository.dart]: @divy-arpit Read about `List.map`, `List.forEach` and their differences

**Comment by werainkhatri** [comment]
Taken over as PR #117

**Pull request closed without merging**